packages/react-server/src/ReactFizzServer.js

```diff
@@ -3281,6 +3373,20 @@ function finishedTask(
         request.completedBoundaries.push(boundary);
       }
 
+      if (enableFloat && parentHoistables) {
+        // We have completed a boundary and need to merge this boundary's Hoistables with the parent
+        // Hoistables for this task. First we remove the fallback Hoistables. If they already flushed
+        // we can't do anythign about it but we don't want to flush them now if unflushed because the fallback
+        // will never show
+        if (boundary.fallbackHoistables) {
+          parentHoistables.fallbacks.delete(boundary.fallbackHoistables);
+        }
+        // Next we merge the boundary Hoistables into the task Hoistables. In the process the boundary assumes
+        // the task Hoistables internal state so later if a child task also completes it will merge with
+        // the appropriate sets
+        mergeHoistables.call(parentHoistables, hoistables);
```

Comment:
This is a common call. Shouldn't use `.call` deopt. Better to split it out into a separate helper function.